data_processing/process_data.py

```diff
@@ -0,0 +1,27 @@
+# This script handles the processing and creation of the interaction matrix
+
+import numpy as np
+
+def create_interaction_matrix(directory_path, num_users, num_movies, files):
+    interaction_matrix = np.zeros((num_users, num_movies), dtype=np.int8)
+    user_mapper = {}
+    current_user_index = 0
+    
+    for filename in files:
+        with open(os.path.join(directory_path, filename), 'r') as file:
+            movie_id = int(file.readline().strip().replace(':', ''))
+            movie_idx = movie_id - 1
+            
+            for line in file:
+                user_id, rating, _ = line.strip().split(',')
+                user_id = int(user_id)
+                rating = int(rating)
+                
+                if rating in (4, 5):
+                    if user_id not in user_mapper:
+                        user_mapper[user_id] = current_user_index
```

Comment:
We might want to pickle and save the user_mapper dictionary as well, so we don't have to write a separate function to calculate it again.

Reply:
On second thought, I'm not sure the user ids are meaningful or useful once we've calculated the matrix.